superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts

```diff
@@ -192,7 +192,7 @@ export const selectNativeIndicatorsForChart = (
     if (!isAffectedByScope) {
       return IndicatorStatus.Unset;
     }
-    if (!column) {
+    if (!column && value.length > 0) {
```

Comment:
`value` not always `array` I think it can be also `string` or `number`, so it will be failed

Reply:
Good point. I believe we should start enforcing the assumption that `value === null` means unset, all else is set.

Reply:
Actually, it's defined as `value: string[]` a few lines above, which feels wrong. Let me look into this.

Reply:
I think it leftovers from the time when we had only select filters

Reply:
I've added a `FilterValue` and replaced everywhere filter values are used (they were both `any` and `string[]`).